src/o365/spfx/commands/project/project-externalize.ts

```diff
@@ -0,0 +1,237 @@
+import commands from '../../commands';
+import Command, {
+  CommandOption, CommandError, CommandAction, CommandValidate
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as path from 'path';
+import * as fs from 'fs';
+import { Utils } from './project-upgrade/';
+import { Project } from './project-upgrade/model';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+import rules = require('./project-externalize/DefaultRules');
+import { BasicDependencyRule } from './project-externalize/rules';
+import { ExternalizeEntry } from './project-externalize/model/ExternalizeEntry';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  outputFile?: string;
+}
+
+class SpfxProjectExternalizeCommand extends Command {
+  private projectVersion: string | undefined;
+  private projectRootPath: string | null = null;
+  private supportedVersions: string[] = [
+    '1.9.1'
+  ];
+  private allFindings: ExternalizeEntry[] = [];
+  public static ERROR_NO_PROJECT_ROOT_FOLDER: number = 1;
+  public static ERROR_NO_VERSION: number = 3;
+  public static ERROR_UNSUPPORTED_VERSION: number = 2;
+  public get name(): string {
+    return commands.PROJECT_EXTERNALIZE;
+  }
+  public get description(): string {
+    return 'Exteranlizes SharePoint Framework project known dependencies';
+  }
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    return telemetryProps;
+  }
+  public action(): CommandAction {
+    const cmd: Command = this;
+    return function (this: CommandInstance, args: CommandArgs, cb: (err?: any) => void) {
+      args = (cmd as any).processArgs(args);
+      (cmd as any).initAction(args, this);
+      cmd.commandAction(this, args, cb);
+    }
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
```

Comment:
Just as suggestion to add `cmd.log('This api is in beta');` or a similar message to notify the user that we are still working on it.